ravenframework/Optimizers/BayesianOptimizer.py

```diff
@@ -0,0 +1,758 @@
+    # Value of the acqusition function post-selection for this iteration
+    toAdd = self._acquFunction.updateSolutionExport()
+    # How close was this sample to the current best?
+    pointDelta = {}
+    # How close to previous sample?
+    recentDelta = {}
+    if self._iteration[traj] > 0:
+      # If this point is the new opt, then it is zero away from itself
+      if acceptable == 'accepted':
+        bestDelta = 0
+
+        for varName in list(self.toBeSampled):
+          newPoint = rlz[varName]
+          # -2 because newPoint is already appended to self._trainingInputs
+          prevPoint = self._trainingInputs[traj][varName][-2]
+          recentDelta[varName] = newPoint - prevPoint
+        prevDelta = np.linalg.norm(self.featurePointToArray(recentDelta))
+      else:
+
+        for varName in list(self.toBeSampled):
+          newPoint = rlz[varName]
+          # -2 because newPoint is already appended to self._trainingInputs
+          prevPoint = self._trainingInputs[traj][varName][-2]
+          bestPoint = self._optPointHistory[traj][-1][0][varName]
+          pointDelta[varName] = newPoint - bestPoint
+          recentDelta[varName] = newPoint - prevPoint
+        bestDelta = np.linalg.norm(self.featurePointToArray(pointDelta))
+        prevDelta = np.linalg.norm(self.featurePointToArray(recentDelta))
+    else:
+      bestDelta = None
+      prevDelta = None
+    toAdd['radiusFromBest'] = bestDelta
+    toAdd['radiusFromLast'] = prevDelta
+    toAdd['solutionValue'] = self._expectedOptVal
+    toAdd['solutionDeviation'] = self._optValSigma
+    toAdd['modelRuns'] = self._evaluationCount
+    return toAdd
+
+  def incrementIteration(self, traj):
+    """
+      Increments the "generation" or "iteration" of an optimization algorithm.
+      The definition of generation is algorithm-specific; this is a utility for tracking only.
+      @ In, traj, int, identifier for trajectory
+      @ Out, None
+    """
+    self._iteration[traj] += 1
+
+  def getIteration(self, traj):
+    """
+      Provides the "generation" or "iteration" of an optimization algorithm.
+      The definition of generation is algorithm-specific; this is a utility for tracking only.
+      @ In, traj, int, identifier for trajectory
+      @ Out, counter, int, iteration of the trajectory
+    """
+    return self._iteration[traj]
+
+  #################################
+  # Model Training and Evaluation #
+  #################################
+  def _setModelBounds(self):
+    """
+      Corrects hyperparameter bounds to account for normalized input space
+      @ In, None
+      @ Out, None
+    """
+    # NOTE This assumes scikitlearn GPR model
+    hyperParamList = self._model.supervisedContainer[0].model.kernel.hyperparameters
+    for hyperParam in hyperParamList:
+      if 'length_scale' in hyperParam.name:
+        hyperBound = hyperParam.name + '_bounds'
+        self._model.supervisedContainer[0].model.kernel.set_params(**{hyperBound:(1e-5,1)})
```

Comment:
I believe this is a good set of bounds without prior knowledge of the function. Since the model is trained on a normalized feature space ([0,1]), length-scales greater than 1 imply that dimension does not change significantly over the entire domain. 1e-5 is what I believe to be a reasonable lower bound as it essentially implies there is no correlation between neighboring points for that dimension.

Reply:
Please add your comments to the code.